src/requests/consumption.js

```diff
@@ -0,0 +1,1059 @@
+				deleteFolderRecursive(currentPath)
+			} else {
+				// Otherwise, delete the file
+				fs.unlinkSync(currentPath)
+			}
+		})
+		// Delete the empty folder
+		fs.rmdirSync(folderPath)
+		console.log(`Folder and its contents deleted: ${folderPath}`)
+	} else {
+		console.log('Folder does not exist:', folderPath)
+	}
+}
+// Function to fetch data information from cloud using downloadable Url
+async function getBaseTemplate(templateUrl) {
+	try {
+		const response = await axios.get(templateUrl)
+		if (response.status === 200) {
+			return {
+				success: true,
+				result: response.data,
+			}
+		} else {
+			throw new Error(`Unexpected response status: ${response.status}`)
+		}
+	} catch (error) {
+		return Promise.reject(new Error(`Failed to fetch base template: ${error.message}`))
+	}
+}
+
+// download file from cloud and convert it into base64
+async function downloadAndConvertToBase64(url) {
+	try {
+		// Download the image file as a binary buffer
+		const response = await axios({
+			url,
+			method: 'GET',
+			responseType: 'arraybuffer', // Ensures we receive raw binary data
+			timeout: 40000,
+			headers: {
+				Connection: 'close', // Ensures the socket is closed after the request
+			},
+		})
+
+		// Convert the binary data to a Base64 string
+		const base64 = Buffer.from(response.data, 'binary').toString('base64')
+
+		// Get the content type (e.g., image/jpeg) from the response headers
+		const contentType = response.headers['content-type']
+
+		// Create the Base64 Data URL
+		const base64DataUrl = `data:${contentType};base64,${base64}`
+
+		return base64DataUrl
+	} catch (error) {
+		console.error('Error downloading or converting file:', error.message)
+		throw error
+	}
+}
+
+// async function createResourceRollouts()
+
+/**
+ * Publish the Program
+ * @name publishProjectTemplates
+ * @param {Object} programData - Program template data
+ * @returns {Object} - Response of Program creation
+ */
+const publishProgram = function (programData) {
+	return new Promise(async (resolve, reject) => {
+		const result = { success: false, templateId: null, error: null }
+		try {
+			// Format the template
+			let formattedTemplate = formatProgramTemplate(programData)
+			if (!formattedTemplate.success) {
+				throw new Error('FAILED_TO_FORMAT_TEMPLATE')
+			}
+
+			let template = formattedTemplate.template
+			const resourceDetailsCreate = template.resourceDetails.map(
+				(resource) => resource.processType == common.ROLLOUT_PROCESS_TYPE_CREATE
+			)
+			const resourceDetailsUpdate = template.resourceDetails.map(
+				(resource) => resource.processType == common.ROLLOUT_PROCESS_TYPE_UPDATE
+			)
+			const programScope = template.scope
+			delete template.resourceDetails
+			let result = {}
+			let programId
+			// Insert the template into the database
+			const programsCollection = mongoDb.collection(COLLECTIONS.PROGRAMS)
+			if (programData.processType == common.ROLLOUT_PROCESS_TYPE_UPDATE) {
```

Comment:
@adithyadinesh0412 can we check published_id is there or not, instead of processType